### ISSUE TITLE
Microgrant -  Test vectors implementation for FEVM: Test Application and Execution Traces

### ISSUE THREAD

**Issue by shamb0** [filecoin-project/devgrants#1392]
### 1. What is your project, and what problem does it solve? (max 100 words)

[Tool for applying test vectors from Ethereum on FEVM](https://github.com/storswiftlabs/devgrants/blob/fvm_ethereum_test/rfps/fvm-ethereum-test-vectors.md).

> *Milestone-02 

1. Port the deliverables of [Milestone-01](https://github.com/filecoin-project/devgrants/issues/1391), to ref-fvm. And execute the test vectors under WASM context.

2. Capture the test report in tracking sheet with the links to execution traces, & reported new issue ID.

3. Long term support for delivered test application.

### 2. Project links

| Deliverables | Links |
| --- | --- |
| Test Application | https://github.com/shamb0/fevm-eth-compliance-test |
| Execution Traces | https://github.com/shamb0/fevm-eth-compliance-test-trace/tree/main/GeneralStateTests |
| Report | [Tracking Sheet](https://docs.google.com/spreadsheets/d/1g0FLXqMUnCs85eGbvT9_0TdToqqKc0yHAGY_XQStZKo/edit?usp=sharing) |
  
### 3. a) How is IPFS, Filecoin, or related technology used in this project?

> *Please refer to proposed design [presentation](https://docs.google.com/presentation/d/1u_-CamlnGZAVuY2ci3JSNnFq51l4X_TH/edit#slide=id.p1)*

###    b) Is this project building with the current microgrants focus area (FVM)? (Yes or No)

> *Yes*

### 4. How will you improve your project with this grant? What steps will you take to meet this objective? (max 200 words)

> *Objective of the task is to perform FEVM Eth compliance test based on standard [Eth Test Vectors](https://github.com/ethereum/tests/tree/develop/GeneralStateTests).*

### 5. If accepted, do you agree to share monthly project updates in this Github issue for 3 months or until the project described here is complete?

> Yes

### 6. Does your proposal comply with our Community Code of Conduct?

> Yes

### 7. Links and submissions

> [PR 953](https://github.com/filecoin-project/builtin-actors/pull/953)
  
### Additional questions:

* For each team member(s), please list name, email, Github account, and role in the project.

| ID | Details |
| --- | --- |
| list name | shamb0 |
| email | r.raajey@gmail.com |
| github | https://github.com/shamb0 |
| Role | Community Contributor |

* How did you learn about our microgrant program?
 
> *Via Slack channel, posed by Maciej Witowski (@maciej)*

* Grant implementation cost?
> *The implementation cost for this microgrant is 5000 $.
We are ok to have this amount paid out in FIL. *

* If your project was created as part of an event or hackathon:
  
> *RFP | https://github.com/storswiftlabs/devgrants/blob/fvmethereumtest/rfps/fvm-ethereum-test-vectors.md*

**Comment by BlocksOnAChain** [comment]
@ErinOCon @realChainLife -  This Microgrant is approved by the FVM core development team. 
- Please proceed with the appropriate steps for approval from your side.
Thanks 🙏 

cc: @maciejwitowski for visibility

**Comment by ErinOCon** [comment]
Hi @shamb0, thank you for your patience with our review process. This microgrant has been approved! We will send an email to confirm payment details. Please note that your final grant report is due within three months and can be submitted [here](https://airtable.com/shrrtSlgdVvrPmAvr). 

Please let us know if you have any questions!

**Comment by shamb0** [comment]
Hi @ErinOCon, thanks for review closure. Sure will complete the final grant report soon.

Can I have your kind support to review and close another related micro grant ticket ? 

https://github.com/filecoin-project/devgrants/issues/1391

**Comment by ErinOCon** [comment]
Hi @shamb0, thank you for checking in on the status of the other microgrant proposal. We will only proceeding with one microgrant at this time, but can consider additional proposals in the future. With this in mind, would you prefer to update the scope of this grant?

**Comment by BlocksOnAChain** [comment]
@shamb0 can we maybe merge 1391 into this one by adding everything you did in a single one?
Sorry for the delays.
@ErinOCon would this be ok with you - if we merge it and add the costs in a single place?

**Comment by shamb0** [comment]
Sure, @Dragan Z, I'm ready to merge all changes into a single pull request.

**Comment by ErinOCon** [comment]
Hi @BlocksOnAChain and @shamb0, the value of the grant amount should remain the same ($5k USD). Once the work is complete, we can consider a new proposal. 

If preferred, the scope of this proposal can be revised to account for the single payment.

**Comment by shamb0** [comment]
@ErinOCon, in my view, the scope of the work is already complete, and I have discussed it with @BlocksOnAChain and @maciejwitowski. We have all agreed to any feature additions or bug fixes at no additional cost.

Here is a short overview of the journey of the work, which took approximately 8 weeks to complete 95% of the task. We estimated the work done to be worth 8K USD. However, since this amount is beyond the upper limit of microgrants budget, @BlocksOnAChain suggested splitting the work into two tasks.

* Task-01 | Ticket (#1391) : Initially, the test application was built for the 'buildin-actors' mock runtime context. After executing approximately 100 test vectors, based on the test results and an initial level of triage, we realized that the test vectors needed to be executed in a WASM runtime instead of the native context. We decided to port the test application to the 'ref-fvm' integration test WASM runtime. So all this initial work are part of the ticket (#1391).

* Task-02 | Ticket (#1392): Captures all the porting efforts to the ref-fvm integration test WASM runtime context, including the corresponding code and a test report of the execution of all test vectors.

@BlocksOnAChain and @maciejwitowski, please share your comments if there is any deviation in our alignment or delivered work.

Please guide me on what I have to do next?

**Comment by ErinOCon** [comment]
Hi @shamb0, thank you for the full summary. I just sent you an email to discuss the best way forward!

**Comment by ErinOCon** [comment]
Hi @shamb0, confirming this proposal has been approved as an Open grant and will be combined with the milestone described [here](https://github.com/filecoin-project/devgrants/issues/1391).
